File: PiBackend/src/main/java/com/coco/pibackend/Entity/AnnonceCollocation.java
package com.coco.pibackend.Entity;

import com.coco.pibackend.Enum.Type_annon_Collocation;
import com.coco.pibackend.Enum.Type_logement;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;
import java.util.Date;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class AnnonceCollocation  implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id_anno_colo;
    private boolean status;
    private String Addresse;
    private String ville;
    private String pays;
    private Date date_disponiblite;
    private int nbrChambre;
    private String meuble;
    private String photos;
    private float cautionnement;
    private String sexe;
    @Enumerated(EnumType.STRING)
    private Type_logement typeLogement;
    private String description;
    private float montantContrubition;
    @Enumerated(EnumType.STRING)
    private Type_annon_Collocation typeAnnonCollocation;
    private int nbrPersonne;
    @ManyToMany(mappedBy = "annonceCollocations")
    private Set<Avis> avis;
    @ManyToOne
    private User user;
    @OneToMany(mappedBy = "annonceCollocation",cascade = CascadeType.ALL)
    @JsonIgnore
    private List<RendezVous>rendezVous;







}

File: PiFrontEnd/src/app/model/AnnonceCollocation.ts
import {user} from '../model/User';
import {Type_logement} from '../model/Type_logement';
import {Type_annon_Collocation} from '../model/Type_annon_Collocation';
export class AnnonceCollocation{
    id_anno_colo!:number;
    addresse!: string;
    ville!: string;
    pays!: string;
    status!: boolean;
    date_disponiblite!: Date;
    nbrChambre!:number;
    meuble!: string;
    photos!: string;
    cautionnement!:number;
    sexe!: string;
    typeLogement!: Type_logement;
    typeAnnonCollocation!: Type_annon_Collocation;
    description!: string;
    montantContrubition!:number;
    nbrPersonne!:number;
    url!:string;
    user!:user;
    Isityours!: boolean;
}
